zipkin/src/main/java/zipkin/storage/InMemorySpanStore.java

```diff
@@ -84,13 +84,8 @@
   /** This supports span lookup by {@link zipkin.Span#traceId lower 64-bits of the trace ID} */
   private final SortedMultimap<Long, Pair<Long>> traceIdToTraceIdTimeStamps =
       new LinkedHashSetSortedMultimap<>(Long::compareTo);
-  /**
-   * This supports span lookup by {@link zipkin.Endpoint#serviceName service name}.
-   *
-   * <p>QueryRequest.limit needs trace ids are returned in timestamp descending order.
-   */
-  private final ServiceNameToTraceIdTimeStamp serviceToTraceIdTimestamp =
-      new ServiceNameToTraceIdTimeStamp();
+  /** This is an index of {@link Span#traceId} by {@link zipkin.Endpoint#serviceName service name} */
+  private final ServiceNameToTraceIds serviceToTraceIds = new ServiceNameToTraceIds();
```

Comment:
big change here is reduced cardinality on the values to service -> traceId vs service -> (traceId, timestamp)

The impact is easier cleanup and a bit less scanning during writes (due to less cleanup there). Should be measurable, but we don't have benchmarks in for this at the moment.